sql/table.cc

```diff
@@ -8479,6 +8484,199 @@ LEX_CSTRING *fk_option_name(enum_fk_option opt)
   return names + opt;
 }
 
+TR_table::TR_table(THD* _thd, bool rw) : thd(_thd)
+{
+  static const LString table_name("transaction_registry");
+  init_one_table(LEX_STRING_WITH_LEN(MYSQL_SCHEMA_NAME),
+                 XSTRING_WITH_LEN(table_name), table_name, rw ? TL_WRITE : TL_READ);
+  open_tables_backup= new Open_tables_backup;
+  if (open_tables_backup)
+    open_log_table(thd, this, open_tables_backup);
+  else
+    my_error(ER_OUT_OF_RESOURCES, MYF(0));
+}
+
+TR_table::~TR_table()
+{
+  if (table)
+    close_log_table(thd, open_tables_backup);
+  delete open_tables_backup;
+}
+
+void TR_table::store(uint field_id, ulonglong val)
+{
+  table->field[field_id]->store(val, true);
+  table->field[field_id]->set_notnull();
+}
+
+void TR_table::store(uint field_id, timeval ts)
+{
+  table->field[field_id]->store_timestamp(ts.tv_sec, ts.tv_usec);
+  table->field[field_id]->set_notnull();
+}
+
+void TR_table::store_data(ulonglong trx_id, ulonglong commit_id, timeval &commit_ts)
```

Comment:
Pass `commit_ts` by value here?

Reply:
Point?

Reply:
Simplicity. No point in a reference here. Not a big issue, though.

Reply:
Agree, it makes additional questions. Fixed.